addons/website_quote/models/sale_quote.py

```diff
@@ -0,0 +1,73 @@
+from openerp import api, fields, models
+
+import openerp.addons.decimal_precision as dp
+
+class SaleQuoteTemplate(models.Model):
+    _name = "sale.quote.template"
+    _description = "Sale Quotation Template"
+
+    name = fields.Char('Quotation Template', required=True)
+    website_description = fields.Html(string='Description', translate=True)
+    quote_line = fields.One2many('sale.quote.line', 'quote_id', string='Quotation Template Lines', copy=True)
+    note = fields.Text(string='Terms and conditions')
+    options = fields.One2many('sale.quote.option', 'template_id', string='Optional Products Lines', copy=True)
+    number_of_days = fields.Integer(string='Quotation Duration', help='Number of days for the validity date computation of the quotation')
+
+    @api.multi
+    def action_open_template(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'target': 'self',
+            'url': '/quote/template/%d' % self.id
+        }
+
+
+class SaleQuoteLine(models.Model):
+    _name = "sale.quote.line"
+    _description = "Quotation Template Lines"
+    _order = 'sequence, id'
+
+    sequence = fields.Integer(help="Gives the sequence order when displaying a list of sale quote lines.", default=10)
+    quote_id = fields.Many2one('sale.quote.template', string='Quotation Template Reference', required=True, ondelete='cascade', index=True)
+    name = fields.Text(string='Description', required=True, translate=True)
+    product_id = fields.Many2one('product.product', string='Product', domain=[('sale_ok', '=', True)], required=True)
+    website_description = fields.Html(string='Line Description', compute='_compute_website_description', translate=True)
+    price_unit = fields.Float(string='Unit Price', required=True, digits_compute=dp.get_precision('Product Price'))
+    discount = fields.Float(digits_compute=dp.get_precision('Discount'))
+    product_uom_qty = fields.Float(string='Quantity', required=True, digits_compute=dp.get_precision('Product UoS'), default=1)
+    product_uom_id = fields.Many2one('product.uom', string='Unit of Measure ', required=True)
+
+    @api.one
+    @api.depends('product_id.product_tmpl_id.quote_description')
+    def _compute_website_description(self):
+        if self.product_id:
+            self.website_description = self.product_id.quote_description or self.product_id.website_description or ''
+
+    @api.onchange('product_id')
+    def on_change_product_id(self):
+        self.price_unit = self.product_id.list_price
+        self.product_uom_id = self.product_id.uom_id.id
+        self.website_description = self.product_id and (self.product_id.quote_description or self.product_id.website_description) or ''
```

Comment:
same here no need to check self.product_id. as if product id not found it's return null recordset